### DISCUSSION TITLE
Queries: Unordered Child Nodes

### DISCUSSION THREAD

**Discussion by mr-martian** [tree-sitter/tree-sitter#3998] [Q&A] [answered]
I have a grammar in which a set of flags can appear in an arbitrary order and I want to query for particular combinations of flags, regardless of order.

For example, matching `(flags (flagA) (flagB) (flagC))`, `(flags (flagC) (flagA) (flagB))`, etc, but requiring that all three be present.

Is there a way to do this?

**Comment by amaanq** [comment, marked as answer]
The query API does not provide this OOTB, you would either have to:

- write a custom predicate to support this (in which the predicate does the validation)
- post-process your match with the validation logic (basically the prior point but without using a custom predicate)
